sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,265 @@
+        - An iterable yielding train/test splits.
+
+        For integer/None inputs, if the estimator is a classifier and ``y`` is
+        either binary or multiclass,
+        :class:`sklearn.model_selection.StratifiedKFold` is used. In all
+        other cases, :class:`sklearn.model_selection.KFold` is used.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    method : string, optional (default='auto')
+        This method will be called on the estimator to produce the output of
+        transform. If the method is ``auto``, will try to invoke, for each
+        estimator, ``predict_proba``, ``decision_function`` or ``predict``
+        in that order.
+
+    n_jobs : int, optional (default=1)
+        Number of jobs to be passed to ``cross_val_predict`` during
+        ``fit_transform``.
+    """
+    def __init__(self, estimator, cv=3, method='auto', n_jobs=1):
+        self.estimator = estimator
+        self.cv = cv
+        self.method = method
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y, **fit_params):
+        """Fit the estimator.
+
+        This should only be used in special situations. Read more in the
+        :ref:`User Guide <stacking_transformer>`.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        **fit_params : parameters to be passed to the base estimator.
+
+        Returns
+        -------
+        self : object
+
+        """
+        self.estimator_ = clone(self.estimator)
+        self.estimator_.fit(X, y, **fit_params)
+        return self
+
+    def _method_name(self):
+        if self.method == 'auto':
+            if getattr(self.estimator_, 'predict_proba', None):
+                method = 'predict_proba'
+            elif getattr(self.estimator_, 'decision_function', None):
+                method = 'decision_function'
+            else:
+                method = 'predict'
+        else:
+            method = self.method
+
+        return method
+
+    def transform(self, *args, **kwargs):
+        """Transform dataset.
+
+        Note that, unlike ``fit_transform()``, this won't return the cross
+        validation predictions. Read more in the
+        :ref:`User Guide <stacking_transformer>`.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape=(n_samples, n_features)
+            Input data to be transformed. Use ``dtype=np.float32`` for maximum
+            efficiency. Sparse matrices are also supported, use sparse
+            ``csr_matrix`` for maximum efficiency.
+
+        Returns
+        -------
+        X_transformed : sparse matrix, shape=(n_samples, n_out)
```

Comment:
is it always a sparse matrix?

shape=(n_samples, n_out) -> shape (n_samples, n_estimators)